pkg/sql/sem/builtins/builtins.go

```diff
@@ -5145,9 +5145,10 @@ func regexpReplace(ctx *tree.EvalContext, s, pattern, to, sqlFlags string) (tree
 						newString.WriteString(s[matchStart:matchEnd])
 					} else {
 						idx := int(to[i] - '0')
-						// regexpReplace expects references to "out-of-bounds" capture groups
-						// to be ignored.
-						if 2*idx < len(matchIndex) {
+						// regexpReplace expects references to "out-of-bounds"
+						// and empty (when the corresponding match indices
+						// are negative) capture groups to be ignored.
+						if 2*idx < len(matchIndex) && matchIndex[2*idx] >= 0 {
```

Comment:
can you capture some of these variables and arithmetic expressions? Currently this expression is a bit of an eye chart.

You could capture 2*idx, and matchIndex[2*idx+1]  maybe.